### PR TITLE
Use common workflows

### PR DESCRIPTION
**Pull request by eric-young**
# Description
Using common workflows defined in dell/common-github-actions
- utilize common workflows
- remove redundant workflows defined in this repo
- added Makefile targets to test workflows (`make check` and/or `make actions`)
- removed local config files for yamllint and golangci-lint as they now reside in the common-github-actions repo

# GitHub Issues
None

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [X] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [n/a] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have maintained backward compatibility

# How Has This Been Tested?
Extensive testing with ACT tool to validate workflows

### CODE REVIEW

File: .github/workflows/common-workflows.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: Workflow
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: ["**"]
+
+jobs:
+
+  # golang statis analysis checks
```
Comment (shaynafinocchiaro) [inline .github/workflows/common-workflows.yaml:10]: Misspelling on static

File: pkg/modules/authorization.go [hunk 2]
```diff
@@ -714,6 +714,7 @@ func authorizationStorageServiceV2(ctx context.Context, isDeleting bool, cr csmv
 		return fmt.Errorf("applying/deleting vault certificates: %w", err)
 	}
 
+	// replicates needs to be a int32
```
Comment (shaynafinocchiaro) [inline pkg/modules/authorization.go:717]: Misspelling on replicas

File: .github/workflows/common-workflows.yaml (hunk 1 above)
Comment (eric-young) [reply .github/workflows/common-workflows.yaml]: resolved, please feel free to use the suggestions with the comments to provide a fix

File: pkg/modules/authorization.go (hunk 2 above)
Comment (eric-young) [reply pkg/modules/authorization.go]: resolved, please feel free to use the suggestions with the comments to provide a fix

**Review by abhi16394 — APPROVED** [review]

**Review by shaynafinocchiaro — APPROVED** [review]

**Pull request merged**